---
title: try
description: Convert a function to an error-first async function
---

import { SourceLinkAndPreview } from '@/components/SourceLinkAndPreview'
import { TestingLinkAndPreview } from '@/components/TestingLinkAndPreview'

## Basic usage

Error-first callbacks were cool. Using mutable variables to hoist state when doing try/catch was not cool.

The `_.try` function let's you wrap a function to convert it to an error-first async function.

```ts
import * as _ from 'radash'

const [err, user] = await _.try(api.users.find)(userId)
```

Note, because `try` is a reserved word this function is aliased as `try` and actually named `tryit` in the source.  

### Currying

You can curry `try` if you like.

```ts
import * as _ from 'radash'

const findUser = _.try(api.users.find)

const [err, user] = await findUser(userId)
```

## Testing

<TestingLinkAndPreview module="async" func="try" />

## Source

<SourceLinkAndPreview module="async" func="try" />
